variables.tf

```
@@ -149,3 +149,8 @@ variable "is_frontend" {
   description = "if set to true, tf will create a WAF enabled application gateway"
   default = "0"
 }
+
+variable "additional_host_names" {
+  type = "string"
```

I think a list type would make more sense, it seems like a more natural fit here

not sure if spaces after commas are an issue but that would prevent that sort of thing.
```suggestion
  type = "list"
```

You could then join it with:
https://www.terraform.io/docs/configuration/interpolation.html#join-delim-list-
```
join(",", var.additional_host_names)
```

---

Agree with Tim here

---

Initially wanted to use a list all the way through however was having type issues at the template level so settled with string. I didn't think about building the string in the module, good shout